src/core.ts

```diff
@@ -355,36 +355,66 @@ export type TrackingConsent = {
   [purpose: string]: boolean | undefined;
 };
 
-/** Tracking purpose consent config with timestamp & auth metadata */
-export type TrackingConsentDetails = {
-  /** Tracking consent config */
-  purposes: TrackingConsent;
-  /**
-   * Was tracking consent confirmed by the user?
-   * If this is false, the consent was resolved from defaults & is not yet confirmed
-   */
-  confirmed: boolean;
-  /** Consent resolution/last-modified timestamp (ISO 8601) */
-  timestamp: string;
-  /** Has the consent been updated (including no-change confirmation) since default resolution */
-  updated?: boolean;
-  /** Whether or not the UI has been shown to the end-user (undefined in older versions of airgap.js) */
-  prompted?: boolean;
-  /** Transparency Consent (TCF) String */
-  tcf?: string;
-  /** US Privacy (USP) String */
-  usp?: string;
-  /** Global Privacy Platform (GPP) String */
-  gpp?: string;
-  /** Consent Manager View State */
-  viewState?: ViewState;
-  /** Airgap Version */
-  airgapVersion?: string;
-  /** Arbitrary metadata that customers want to be associated with consent state */
-  metadata?: unknown;
-  /** When the metadata was last updated */
-  metadataTimestamp?: string;
-};
+export const TrackingConsentDetails = t.intersection([
+  t.type({
+    /**
+     * Was tracking consent confirmed by the user?
+     * If this is false, the consent was resolved from defaults & is not yet confirmed
+     */
+    confirmed: t.boolean,
+    /** Consent resolution/last-modified timestamp (ISO 8601) */
+    timestamp: t.string,
+  }),
+  t.partial({
+    /** Has the consent been updated (including no-change confirmation) since default resolution */
+    updated: t.boolean,
+    /** Tracking consent config */
+    purposes: t.record(t.string, t.boolean),
```

Comment:
This should be backwards compatible with this caveat, that `purposes` is now optional. The reasoning is that now that consent can be Airgap, USP, TCF, GPP, or any combination of those, no one of them should be required. If this is a problem, @eligrey , we can go keep it required (with the implication being that the TCF UI will have to send an empty object for `purposes` is Airgap is not present).

Reply:
Can we make it required for now? TCF and GPP isn't even supported yet, and USP + airgap purposes are actually co-occuring.

Reply:
airgap.js only cares about the types, it doesn't use the codec, so we could also do something like:

```
export type OptionalPurposeTrackingConsentDetails = t.TypeOf<typeof TrackingConsentDetails>;
export type TrackingConsentDetails = Requirize<OptionalPurposeTrackingConsentDetails, 'purposes'>;
```

Reply:
`purposes` from the `getConsent()` API are required so we should not change this. If we must make purposes optional then there should be a different type for that use case (which gets requirized in for TrackingConsentDetails)

Reply:
I think for simplicity I'll just make it required now, but good to know we have that option to `Requirize` if we need to make it optional on the backend